### PR TITLE
feat: Add an API to get an organization member

### PR DESCRIPTION
**Pull request by yu-iskw**
Closes: https://github.com/lightdash/lightdash/issues/5914

**Update @owlas**

You can now test with:

```
### Login
POST http://localhost:8080/api/v1/login
Content-Type: application/json

{
    "email": "demo@lightdash.com",
    "password": "demo_password!"
}

### Get all members
GET http://localhost:8080/api/v1/org/users

### Get a member by uuid
GET http://localhost:8080/api/v1/org/users/b264d83a-9000-426a-85ec-3f9c20f368ce

### Get a member by email address
GET http://localhost:8080/api/v1/org/users/emails/demo@lightdash.com

### Get a member that doesn't exist
GET http://localhost:8080/api/v1/org/users/emails/notexists

### Get a member that doesn't exist
GET http://localhost:8080/api/v1/org/users/notexists

### Get a member from another org
GET http://localhost:8080/api/v1/org/users/57cd4548-cbe3-42b3-aa13-97821713e307
```

### Description:
I would like to add an API to get a member's profile. Although we already have the API to get all members of an organization, it wouldn't be efficient if we want to get just one.

In my opinion, there are two use cases to get an user. First, we want to get an user by the UUID of an user. Second, we want to get an user by email.

We consider if we implement a terraform provider for Lightdash. We may want to get a user by the UUID, if we create a terraform data source. Meanwhile, we may want to get an user by the email, if we create a terraform resource. That's why the API I implemented supports either of UUID and email.

```terraform
data "lightdash_organization_member" "sample_member" {
  uuid = "xxxxx-xxxx-xxxx"
}

resource "lightdash_organization_member" "sample_member_02" {
  email = "sample-member-02@test.com"
  role = ...
}
```

### CODE REVIEW

File: packages/backend/src/controllers/organizationController.ts [hunk 1]
```diff
@@ -174,6 +175,41 @@ export class OrganizationController extends Controller {
...
+     * @param req express request
+     * @param body the userUuid or email of the user to get. Either userUuid or email must be provided.
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @Post('/user')
+    @OperationId('ListOrganizationMembers')
+    async getOrganizationMember(
+        @Request() req: express.Request,
+        @Body() body: OrganizationMemberProfileGet,
+    ): Promise<ApiOrganizationMemberProfile> {
```
Comment (yu-iskw) [inline packages/backend/src/controllers/organizationController.ts:189]: We can also separate the methods to get an user's profile by UUID and email. Which do you prefer?

**Comment by owlas** [comment]
Moving PR here so that internal tests trigger: https://github.com/lightdash/lightdash/pull/6288

**Comment by yu-iskw** [comment]
Alright.

**Comment by yu-iskw** [comment]
https://github.com/lightdash/lightdash/pull/6245#issuecomment-1632612201

> GET /org/users/{userUuid} - which follows the existing structure
> 
> and GET /org/users/emails/{email} - for the cases when we want to get by email.

That looks good to me. Thank you for your follow-up.

**Comment by ZeRego** [comment]
@all-contributors add @yu-iskw for code and ideas

**Pull request closed without merging**